engine/src/main/java/com/datatorrent/stram/plan/physical/PhysicalPlan.java

```diff
@@ -979,6 +998,14 @@ public void deployChanges() {
     this.undeployOpers.removeAll(newOpers.keySet());
```

Comment:
You can add the following before this line (979) and then remove everything added new below:
```
// redeploy dependencies of the new operators excluding the new operators themselves
this.undeployOpers.addAll(getDependents(newOpers.keySet()));
```